### PR TITLE
feat: review vasp modal

### PR DESCRIPTION
**Pull request by elysee15**
This PR adds the feature that allows an admin to review a vasp registration

Fixes SC-2661

### CODE REVIEW

**Comment by bbengfort** [comment]
@elysee15 I'd like to try to merge this tomorrow before the release. After you rebase sc-2555, could you rebase this PR from that branch so that I can easily review and merge if possible?

File: web/gds-admin-ui/src/redux/selectors/vasp-details.js [hunk 1]
```diff
@@ -6,4 +6,4 @@ export const getVaspDetails = createSelector(vaspDetailsState, state => state.da
 export const getVaspDetailsLoadingState = createSelector(vaspDetailsState, state => state.loading)
 export const getVaspDetailsErrorState = createSelector(vaspDetailsState, state => state.error)
 export const getTrisaDetailsErrorState = createSelector(vaspDetailsState, state => state.trisaError)
-export const getIvmsErrorState = createSelector(vaspDetailsState, state => state.ivmsError)
\ No newline at end of file
+export const getIvmsErrorState = createSelector(vaspDetailsState, state => state.ivmsError)
```
Comment (bbengfort) [inline web/gds-admin-ui/src/redux/selectors/vasp-details.js:9]: I honestly stared at these two lines of code during the rebase and I can't see what the difference is between them. Just the newline I guess?

File: web/gds-admin-ui/src/pages/app/details/BasicDetails/components/BasicDetailsDropdown.js [hunk 2]
```diff
@@ -42,9 +43,18 @@ const BasicDetailsDropDown = ({ isNotPendingReview, vasp }) => {
                 <i className="dripicons-dots-3"></i>
             </Dropdown.Toggle>
             <Dropdown.Menu>
-                <Dropdown.Item data-testid="reviewItem" disabled={isNotPendingReview()}>
-                    <i className="mdi mdi-card-search me-1"></i>Review
-                </Dropdown.Item>
+                <Modal>
+                    <ModalOpenButton>
+                        <Dropdown.Item data-testid="reviewItem" disabled={isNotPendingReview()}>
```
Comment (bbengfort) [inline web/gds-admin-ui/src/pages/app/details/BasicDetails/components/BasicDetailsDropdown.js:48]: Right now we have two methods of detecting if a review is possible; `isNotPendingReview` and a 404 from `GET /v2/vasps:/id/review`; these seem to be conflicting with each other. I'll create a story to specify how these should interact.

**Review by bbengfort — APPROVED** [review]
@elysee15 looks great; I wasn't able to test thoroughly, but I'd like to get this merged so we can get it out with the deployment today. I'll create stories for any minor hiccups we notice!

File: web/gds-admin-ui/src/redux/selectors/vasp-details.js (hunk 1 above)
Comment (elysee15) [reply web/gds-admin-ui/src/redux/selectors/vasp-details.js:9]: Perhaps it is due to rebasing and conflits resolving

**Pull request merged**